static/src/javascripts/projects/common/modules/social/pinterest.js

```diff
@@ -1,29 +1,31 @@
 // @flow
+
 import fastdom from 'fastdom';
-import bean from 'bean';
-import $ from 'lib/$';
 import { loadScript } from 'lib/load-script';
 
 const launchOverlay = (event: Event): void => {
     event.preventDefault();
 
     const scriptUrl = 'https://assets.pinterest.com/js/pinmarklet.js';
     const cachePurge = new Date().getTime();
+    const images = document.querySelectorAll(
+        'img:not(.gu-image):not(.responsive-img)'
+    );
 
-    $('img:not(.gu-image):not(.responsive-img)').each(img => {
-        fastdom.write(() => {
-            $(img).attr('data-pin-nopin', 'true');
+    fastdom.write(() => {
+        images.forEach(img => {
+            img.setAttribute('data-pin-nopin', 'true');
         });
     });
 
     loadScript(`${scriptUrl}?r=${cachePurge}`);
 };
 
 const initPinterest = (): void => {
-    const buttons = $('.social__item--pinterest');
+    const buttons = document.querySelectorAll('.social__item--pinterest');
 
-    buttons.each(el => {
-        bean.on(el, 'click', launchOverlay);
+    [...buttons].forEach(el => {
```

Comment:
@SiAdcock @katebee I think you are right - for the last couple of weeks I was more in favor of convertig every NodeList into an array, to avoid side effects, but flow can really resolve the issue ...

Will do the proposed change ..